pkg/controllers/ipaddressallocation/ipaddressallocation_controller.go

```diff
@@ -109,8 +105,10 @@ func (r *IPAddressAllocationReconciler) Reconcile(ctx context.Context, req ctrl.
 	log.Info("reconciling IPAddressAllocation CR", "IPAddressAllocation", req.NamespacedName)
 	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerSyncTotal, MetricResType)
 	if err := r.Client.Get(ctx, req.NamespacedName, obj); err != nil {
-		log.Error(err, "unable to fetch IPAddressAllocation CR", "req", req.NamespacedName)
-		return resultNormal, client.IgnoreNotFound(err)
+		if client.IgnoreNotFound(err) == nil {
+			return r.handleDeletion(ctx, req, obj)
```

Comment:
When we hit the not found error, obj will be empty. You may need to find all the IPAddressAllocation by name and skip those IPAddressAllocation associated with the existing CR, which is created with the same name after the last deletion. Maybe refer to the other finalizer removal?

Reply:
Yes, you are right, Thanks~~